six/modules/c++/six/source/NITFWriteControl.cpp

```diff
@@ -866,11 +866,14 @@ void NITFWriteControl::save(
                 iWriter.setWriteCaching(1);
 
                 nitf::ImageSource iSource;
-                const size_t bandSize = numCols * info.getData()->getNumRows();
+                const NITFSegmentInfo segmentInfo = imageSegments[jj];
+                const size_t bandSize = numCols * segmentInfo.numRows;
 
                 for (size_t chan = 0; chan < numChannels; ++chan)
                 {
-                    nitf::MemorySource ms(imageData[i], bandSize,
+                    nitf::MemorySource ms(imageData[i] +
```

Comment:
OK no prob - just doing it via the pointer arithmetic is fine.  I'll take a look at the rest of the changes here shortly.